esp_idf_panic_decoder/pc_address_decoder.py

```diff
@@ -23,50 +30,136 @@ def __init__(
         self.toolchain_prefix = toolchain_prefix
         self.elf_files = elf_file if isinstance(elf_file, list) else [elf_file]
         self.rom_elf_file = rom_elf_file
-        self.pc_address_buffer = b''
         self.pc_address_matcher = [PcAddressMatcher(file) for file in self.elf_files]
-        if rom_elf_file is not None:
-            self.rom_pc_address_matcher = PcAddressMatcher(rom_elf_file)
+        if self.rom_elf_file:
+            self.pc_address_matcher.append(PcAddressMatcher(self.rom_elf_file))
+
+    def decode_address(self, line: bytes):
+        """
+        Decode possible addresses in a line, returning a human-readable string.
+        """
+        decoded = self.decode_addresses(line)
+        if not decoded:
+            return ''
+
+        def format_trace_entry(t: PcAddressLocation):
+            if t.path == 'ROM':
+                return f'{t.fn} in ROM'
+            else:
+                return f'{t.fn} at {t.path}' + (f':{t.line}' if t.line else '')
 
-    def decode_address(self, line: bytes) -> str:
-        """Decoded possible addresses in line"""
-        line = self.pc_address_buffer + line
-        self.pc_address_buffer = b''
         out = ''
-        for match in re.finditer(ADDRESS_RE, line.decode(errors='ignore')):
-            num = match.group()
-            address_int = int(num, 16)
-            translation = None
-
-            # Try looking for the address in the app ELF files
-            for matcher in self.pc_address_matcher:
-                if matcher.is_executable_address(address_int):
-                    translation = self.lookup_pc_address(num, elf_file=matcher.elf_path)
-                    if translation is not None:
-                        break
-            # Not found in app ELF file, check ROM ELF file (if it is available)
-            if translation is None and self.rom_elf_file is not None and \
-            self.rom_pc_address_matcher.is_executable_address(address_int):
-                translation = self.lookup_pc_address(num, is_rom=True, elf_file=self.rom_elf_file)
-
-            # Translation found either in the app or ROM ELF file
-            if translation is not None:
-                out += translation
+        for addr, trace in decoded:
```

Comment:
Could you please put into a comment an example what we are processing? This will help understand the algorithm below.